tests/tests_drampower/interface/test_interface_ddr5.cpp

```diff
@@ -0,0 +1,329 @@
+    EXPECT_EQ(stats.readDQSStats.ones_to_zeroes, DQS_zeros_to_ones);
+    EXPECT_EQ(stats.readDQSStats.zeroes_to_ones, DQS_ones_to_zeros);
+}
+
+TEST_F(DDR5_WindowStats_Tests, Pattern_1) {
+    runCommands(test_patterns[1]);
+
+    SimulationStats stats = ddr->getStats();
+
+    EXPECT_EQ(stats.writeBus.ones, 16);
+    EXPECT_EQ(stats.writeBus.zeroes, 384);
+    EXPECT_EQ(stats.writeBus.ones_to_zeroes, 16);
+    EXPECT_EQ(stats.writeBus.zeroes_to_ones, 16);
+
+    EXPECT_EQ(stats.readBus.ones, 0);
+    EXPECT_EQ(stats.readBus.zeroes, 400);
+    EXPECT_EQ(stats.readBus.ones_to_zeroes, 0);
+    EXPECT_EQ(stats.readBus.zeroes_to_ones, 0);
+
+    EXPECT_EQ(stats.commandBus.ones, 23);
+    EXPECT_EQ(stats.commandBus.zeroes, 341);
+    EXPECT_EQ(stats.commandBus.ones_to_zeroes, 21);
+    EXPECT_EQ(stats.commandBus.zeroes_to_ones, 21);
+}
+
+TEST_F(DDR5_WindowStats_Tests, Pattern_2) {
+    runCommands(test_patterns[2]);
+
+    SimulationStats stats = ddr->getStats();
+
+    EXPECT_EQ(stats.writeBus.ones, 16);
+    EXPECT_EQ(stats.writeBus.zeroes, 464);
+    EXPECT_EQ(stats.writeBus.ones_to_zeroes, 16);
+    EXPECT_EQ(stats.writeBus.zeroes_to_ones, 16);
+
+    EXPECT_EQ(stats.readBus.ones, 1);
+    EXPECT_EQ(stats.readBus.zeroes, 479);
+    EXPECT_EQ(stats.readBus.ones_to_zeroes, 1);
+    EXPECT_EQ(stats.readBus.zeroes_to_ones, 1);
+
+    EXPECT_EQ(stats.commandBus.ones, 31);
+    EXPECT_EQ(stats.commandBus.zeroes, 403);
+    EXPECT_EQ(stats.commandBus.ones_to_zeroes, 28);
+    EXPECT_EQ(stats.commandBus.zeroes_to_ones, 28);
+}
+
+// Tests for power consumption (given a known SimulationStats)
+class DDR5_Energy_Tests : public ::testing::Test {
+   public:
+    DDR5_Energy_Tests() {
+        std::ifstream f(std::string(TEST_RESOURCE_DIR) + "ddr5.json");
+
+        if (!f.is_open()) {
+            std::cout << "Error: Could not open memory specification" << std::endl;
+            exit(1);
+        }
+
+        json data = json::parse(f);
+        spec = MemSpecDDR5{data["memspec"]};
+
+        t_CK = spec.memTimingSpec.tCK;
+        voltage = spec.memPowerSpec[MemSpecDDR5::VoltageDomain::VDDQ].vXX;
+
+        // Change impedances to different values from each other
+        spec.memImpedanceSpec.R_eq_cb = 2;
+        spec.memImpedanceSpec.R_eq_ck = 3;
+        spec.memImpedanceSpec.R_eq_dqs = 4;
+        spec.memImpedanceSpec.R_eq_rb = 5;
+        spec.memImpedanceSpec.R_eq_wb = 6;
+
+        spec.memImpedanceSpec.C_total_cb = 2;
+        spec.memImpedanceSpec.C_total_ck = 3;
+        spec.memImpedanceSpec.C_total_dqs = 4;
+        spec.memImpedanceSpec.C_total_rb = 5;
+        spec.memImpedanceSpec.C_total_wb = 6;
+
+        io_calc = std::make_unique<InterfaceCalculation_DDR5>(spec);
+    }
+
+    MemSpecDDR5 spec;
+    double t_CK;
+    double voltage;
+    std::unique_ptr<InterfaceCalculation_DDR5> io_calc;
+};
+
+TEST_F(DDR5_Energy_Tests, Parameters) {
+    ASSERT_TRUE(t_CK > 0.0);
+    ASSERT_TRUE(voltage > 0.0);
+}
+
```

Comment:
@lsteiner-tukl please pay special attention to the 4 tests below, the goal is to validate the formula being used for the calculation, i.e. our understanding of the termination scheme